src/forms/configure-admin-account.ts

```diff
@@ -58,14 +51,13 @@ export async function opsGenieConfigForm(call: AppCallRequest): Promise<AppForm>
 
 export async function opsGenieConfigSubmit(call: AppCallRequest): Promise<string> {
     const mattermostUrl: string = call.context.mattermost_site_url!;
-    const botAccessToken: string = call.context.bot_access_token!;
+    const accessToken: string = call.context.acting_user_access_token!;
+    const oauth2: Oauth2App = call.context.oauth2 as Oauth2App;
     const values: AppCallValues = <any>call.values;
     const i18nObj = configureI18n(call.context);
-    const actingUser: AppActingUser = call.context.acting_user!;
-
-    if (!isUserSystemAdmin(actingUser)) {
-        throw new Exception(ExceptionType.MARKDOWN, i18nObj.__('forms.configure-admin.system-admin'));
-    }
+    const linkEmailAddress: boolean = typeof oauth2.data?.settings?.link_email_address === 'boolean' ?
+        oauth2.data?.settings.link_email_address :
+        true;
```

Comment:
We are restricting the option to view Configs only to System Admins. However we have removed the restriction while submitting OpsGenie config to all OpsGenie members. This does not makes sense. We should add these restrictions again and restrict view & update configurations only to System Admins.

Reply:
We removed the validation because it was moved to middleware section, but I will also add it to the flow again, so we can have double validation on that

Reply:
Done, this was fixed in [f990168](https://github.com/liz-segura98/mattermost-app-opsgenie/pull/1/commits/f9901682b0106aa464f620e3f35761d3a051d273)